### PR TITLE
Move resume() from BaseAudioContext to its subclasses

### PR DESCRIPTION
**Pull request by hoch**
PTAL for the attempt. There might have more implication than the changes in this PR, but this is a starting point.

Fixes: #1669

***
<a href="https://pr-preview.s3.amazonaws.com/hoch/web-audio-api/pull/1738.html" title="Last updated on Sep 4, 2018, 10:48 PM GMT (eb612fa)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WebAudio/web-audio-api/1738/c05ff33...hoch:eb612fa.html" title="Last updated on Sep 4, 2018, 10:48 PM GMT (eb612fa)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1685,6 +1617,76 @@ Methods</h4>
...
+
+			3. If the {{AudioContext}} is not <a>allowed to
+				start</a>, append <var>promise</var> to
+				<a>pendingResumePromises</a> and abort these steps, returning
+				<var>promise</var>.
+
+			4. Set the <em>control thread state</em> flag on the
+				{{AudioContext}} to <code>running</code>.
+
+			5. <a>Queue a control message</a> to resume the {{AudioContext}}.
```
Comment (rtoy) [inline index.bs]: Not sure what to do here, but the link to Queue a control message talks adding a message to the queue of an AudioContext.  Maybe that section should be changed to BaseAudioContext? Or maybe this needs to be fixed in a completely separate PR.

Comment (rtoy) [reply index.bs]: Please file an issue on this. Otherwise, lgtm except for a lot of duplicated text which I'm not really sure how to fix that.

File: index.bs [hunk 2]
```diff
@@ -1378,21 +1309,22 @@ enum AudioContextLatencyCategory {
 </table>
 </div>
 
-<pre class="idl">
+<xmp class="idl">
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: This is a part of bikeshed: https://tabatkins.github.io/bikeshed/#xmp  I think this is the recommended way in bikeshed to do this so you don't have to escape everything in IDL blocks.

[the remark this answers is not in the document]
Comment (padenot) [reply index.bs]: huh ok.
